### ISSUE TITLE
[BUG] Static `DD_TAGS` not sent to Vector

### ISSUE THREAD

**Issue by smitthakkar96** [DataDog/datadog-agent#15128]
**Agent Environment**

``` shell
Agent 7.41.1 - Commit: 4f39b9e - Serialization version: v5.0.39 - Go version: go1.18.9
```

Tested in Kubernetes and Docker Compose

**Describe what happened:**

Tags set using env var `DD_TAGS` not sent to Vector but when the DD agent directly communicates with intake API the tags are sent correctly.

**Note:** `podLabelAsTags` and tags from other sources are present in `ddtags`

**Describe what you expected:**

Static tags to be present in `ddtags`

**Steps to reproduce the issue:**

Thread in Vector discord: https://discord.com/channels/742820443487993987/1064935290629804163

### Datadog Agent

``` yaml
datadog:
  tags:
    - kubernetescluster:staging-eu-1-v121-green
```

### Vector

``` yaml
  data_dir: /vector-data-dir
  api:
    enabled: true
    address: 127.0.0.1:8686
  sources:
    internal_logs:
      type: internal_logs
    internal_metrics:
      type: internal_metrics
    datadog_agents:
      type: datadog_agent
      address: "[::]:8080"
      multiple_outputs: false
  transforms:
    remap_logs_for_datadog:
      type: remap
      inputs:
        - datadog_agents
      source: |
        # Parse the received .ddtags field so we can more easily access the contained tags
        .ddtags = parse_key_value!(.ddtags, key_value_delimiter: ":", field_delimiter: ",")
        .ddtags.sender = "vector"
        .ddtags.vector_aggregator = get_hostname!()
        # Re-encode Datadog tags as a string for the `datadog_logs` sink
        .ddtags = encode_key_value(.ddtags, key_value_delimiter: ":", field_delimiter: ",")
        # Datadog Agents pass a "status" field that is stripped when ingested
        del(.status)

  sinks:
    datadog_metrics:
      type: datadog_metrics
      inputs:
      - internal_metrics
      default_api_key: "${DATADOG_API_KEY}"
      region: eu
      buffer:
        type: disk
        max_size: 18000000000

    datadog_logs:
      type: datadog_logs
      inputs:
        - remap_logs_for_datadog
        - internal_logs
      default_api_key: "${DATADOG_API_KEY}"
      region: eu
      compression: gzip
      buffer:
        type: disk
        max_size: 18000000000

```

When logs are sent to Datadog `kubernetescluster:staging-eu-1-v121-green` will be missing.

Alternatively use `docker-compose.yml`

``` yaml
version: '3'
services:
  agent:
    image: "datadog/agent:7.41.1"
    environment:
      - DD_API_KEY=${DD_API_KEY}
      - DD_HOST=api.datadoghq.eu
      - DD_APM_ENABLED=true
      - DD_LOGS_ENABLED=true
      - DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true
      - DD_PROCESS_AGENT_ENABLED=true
      - DD_DOCKER_LABELS_AS_TAGS={"my.custom.label.team":"sre-team"}
      - DD_TAGS='env:development'
      - DD_VECTOR_LOGS_ENABLED=true
      - DD_VECTOR_LOGS_URL=http://vector:9126
    ports:
      - "8126:8126"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /proc/:/host/proc/:ro
      - /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
    labels:
      com.datadoghq.ad.logs: '[{"source": "agent", "service": "agent"}]'
  vector:
    image: timberio/vector:0.23.0-debian
    environment:
      - DD_API_KEY=${DD_API_KEY}
    volumes: 
      - ${PWD}:/etc/vector
    ports:
      - 8125:8125
      - 9126:9126
  logger:
    image: chentex/random-logger:latest
    command:
      - "100"
      - "400"
```

**Note:** If you use above docker-compose you will notice `env:development` is missing

Also confirmed by logging the payload from DD Agent to Vector using [linkerd-debug-sidecar](https://linkerd.io/2.12/tasks/using-the-debug-container/)

If using linkerd-debug sidecar run `k exec -it vector-0 -n vector -c linkerd-debug -- tshark -i any -f "tcp" -V -Y "(tcp.srcport == 4143 and tcp.dstport == 50416) or tcp.port == 8080" -c 10 -T fields -T json` to see untruncated payload coming to Vector

**Comment by vickenty** [comment]
Hi @smitthakkar96, thanks for using Datadog.

This is the expected behavior. To reduce amount of data transferred, static tags are not sent with the individual log messages. The tags are sent separately and are applied to log messages when the messages are ingested.

If you have further questions about this or need help finding a solution to your problem, please contact [Datadog support](http://docs.datadoghq.com/help/) and and send them a [flare](https://docs.datadoghq.com/agent/troubleshooting/#send-a-flare) demonstrating the issue.

**Comment by demolitionmode** [comment]
Hi @vickenty, is there a way that we can force the tags to be sent to Vector? I'd like to send logs to other destinations too (eg. our SIEM solution), but without these tags a lot of context is being lost. This also goes for the tags collected from an EC2 instance using `collect_ec2_tags: true` in `agents.customAgentConfig` helm value.

**Comment by vickenty** [comment]
@demolitionmode, to get help with configuring the Agent please reach out to [Datadog support](http://docs.datadoghq.com/help).

**Comment by seanlowcy77** [comment]
@demolitionmode did Datadog support get back to u regarding this?

**Comment by andreilee** [comment]
@vickenty Could you re-open this issue? I think it's clear that there's more interest in this discussion than just a single person. Leaving it here in a public forum if this info is not documented helps us all.

**Comment by demolitionmode** [comment]
> @demolitionmode did Datadog support get back to u regarding this?

I hadn't had the time to raise this before, but I have raised it today; I'll let you know what they respond with (as long as they're happy with the information being public of course 😄)

**Comment by demolitionmode** [comment]
TL;DR: it's not possible, and isn't currently scheduled for development.

They say the best workaround is to use their hosted pipelines; not really what I'd like to hear given that the point of using Vector is to not need to ship every single log to Datadog 😞
